Makefile.rules

```diff
@@ -389,13 +388,16 @@ gap$(EXEEXT): libgap.la cnf/GAP-LDFLAGS cnf/GAP-LIBS cnf/GAP-OBJS
 
 else
 
+all: libgap.so
+libgap.so: symlinks libgap.la
+	$(QUIET_LINK)$(LINK) $(GAP_LDFLAGS) -shared $(OBJS) $(GAP_LIBS) -o $@
```

Comment:
Disclaimer: the following is not meant to block this PR as such, just tries to clarify some background.

When I just looked at the above lines, I wondered again about the relation between libgap.so and libgap.la. There is now a dependency here -- but there seems no strict technical reason.

Moreover, on Mac OS X, the suffix .so is wrong, it should be .dylib. And of course on Windows one would expect gap.dll or so. But on Windows, we technically *already* build GAP into a DLL, as we need that, in order to be able to support kernel extensions at all.

However, all this has an "easy" solution: libgap.la *already builds a shared library*. That's what libtool .la files are for: they abstract over static/shared libraries in a portable way. Building `libgap.la` creates `.libs/libgap.dylib` for me on Mac OS X. Specifically:
```
$ ls .libs/
libgap.0.dylib  libgap.dylib  libgap.la  libgap.lai
```
So it already takes care of things like file extensions.

We abuse this for the Cygwin builds:
```
bin/$(GAPARCH)/gap.dll: symlinks libgap.la
	cp .libs/cyggap-0.dll $@  # FIXME: HACK to support kernel extensions
```

But that' a bad HACK, which I only added to make things work quickly. In reality, one should not directly access files in `.libs`, but rather use the `libtool` script to use these files, e.g. to link against them, or to install them.

Indeed, installing a `.la` may cause libtool to relink it (because the rpaths and other settings that one needs to use the library may differ between when using it inside a build directory during development; and when using it as a system wide installed library).

This all means that this `libgap.so` target should be temporary at best. What we should do instead, though, depends a bit on how `libgap.so` is supposed to be used: are people right now linking against it by specifying `-L/path/to/GAPROOT -lgap`? If so, and if they are using GNU libtool as well, they are better of specifying `/path/to/GAPROOT/libgap.la`. If so, but they are not using GNU libtool, it becomes a bit tricky.
Of course, if instead one requires people to do a `make install` from within GAP (which is not fully implemented, in particular the libgap part...),  then all that goes away... 

All in all, the best long term solution probably is to finally implement `make install`, or at least enough of it to make libgap usable by clients. I always planned to work on that eventually (but demand wasn't exactly high), and would be happy to collaborate on it, too.

Reply:
The way I use `libgap` at the moment is by 

```
(define-ffi-definer define-gap
  (ffi-lib "/home/makx/git/gap/master/.libs/libgap.so"))
```

in Racket (well knowing that this is of course at best a hack, too). If libgap.so were installed, I'd probably be able to just use `ffi-lib "gap". 

I do not require header files, but I have to give `GAP_Initialize` the path to the GAP library. I believe this is mostly how SageMath uses libgap as well.

The reason the `.so` target is there is because I apparently still did not understand libtool. I did remember that the `.la` file is all we need, and that the `.so` file just appeared for me in `.libs`. 

Happy to learn how to do the `make install` so that enough people are happy (but for that we need people to *use* it).